pkg/verify/verify.go

```diff
@@ -193,16 +211,21 @@ func (v *releaseVerifier) Verify(ctx context.Context, releaseDigest string) erro
 	})
 	if err != nil {
 		klog.V(4).Infof("Failed to retrieve signatures for %s (should never happen)", releaseDigest)
+		errs = append(errs, err)
 		return err
 	}
 
 	if len(remaining) > 0 {
-		if klog.V(4).Enabled() {
-			for k := range remaining {
-				klog.Infof("Unable to verify %s against keyring %s", releaseDigest, k)
-			}
+		remainingKeyRings := make([]string, 0, len(remaining))
+		for k := range remaining {
+			remainingKeyRings = append(remainingKeyRings, k)
 		}
-		return fmt.Errorf("unable to locate a valid signature for one or more sources")
+		err := &wrapError{
+			msg: fmt.Sprintf("unable to verify %s against keyrings: %s", releaseDigest, strings.Join(remainingKeyRings, ", ")),
```

Comment:
Looks like we are logging the keys too. I am wondering if it is fine from security point of view>

Reply:
we're logging the keyring name, which is already public in [our GitHub source][1].  The only secret keys here are internal to Red Hat's build-time release signing.  By the time we get out to clusters verifying signatures, it's all public names and public keys.

[1]: https://github.com/openshift/cluster-update-keys/blob/master/manifests.rhel/0000_90_cluster-update-keys_configmap.yaml#L6